src/utilities.cpp

```diff
@@ -431,6 +421,11 @@ ModelPtr owningModel(const EntityPtr &entity)
     return model;
 }
 
+ComponentPtr owningComponent(const VariablePtr &variable)
+{
+    return std::dynamic_pointer_cast<Component>(variable->parent());
+}
+
```

Comment:
Isn't it cleaner to have the `variable->parent() `return a `ComponentPtr`?  The only reason to have the `owningModel` as a separate function for components is because sometimes the parent of a component is a `Model` and sometimes a `Component`?

Reply:
A `Variable` is a `NamedEntity`, which itself is an `Entity`, and `Variable::parent()` is actually `Entity::parent()`, so it returns an `EntityPtr`. However, I guess we could override `Entity::parent()` and have `Variable::parent()` that returns a `ComponentPtr`, but if so then we probably ought to do something similar with `Component` (from which `Model` inherits; this has always baffled me, but that's another debate) and have `Component::parent()` return a `ComponentPtr` and `Component::modelParent()` return a `ModelPtr`

Views? @hsorby, @nickerso?

Reply:
I like the parent concept relating to entities. If anything it would be a new method `Variable::component()`...

Reply:
Ok, so we could have `Variable::component()` and `Component::model()`, and ditch `owningModel()` and `owningComponent()`? I think I would be ok with that. However, if we were to go ahead with that, then we should probably create a new issue for it.

Reply:
> I like the parent concept relating to entities. 
@nickerso  It seems that whenever we use it we have to cast/alter/do something else to it?  What's its value on its own?

Reply:
Yes, 99% of cases, you will need to cast it, which I find it a PITA, hence we have utilities like `owningModel()` and `owningComponent()`, which is more like writing C code rather than C++, but... meh.

Reply:
I understand the desire to use inherited classes, but in this case the inheritance causes more work than if they were defined locally: components having both a ::model() and ::component() class, and variables having a ::component() class.  If inheritance is a must, why not reuse the component() one in the Entity class and add a model() class to the Component!

Reply:
Resolving in the tentative hope that this will be addressed in #615